core/trino-spi/src/main/java/io/trino/spi/type/Decimals.java

```diff
@@ -287,6 +289,37 @@ public static BigDecimal rescale(BigDecimal value, DecimalType type)
         return value;
     }
 
+    public static long rescaleWithRounding(long value, int fromScale, int toScale)
+    {
+        if (toScale < fromScale) {
+            return shortToShortCast(
+                    value,
+                    MAX_SHORT_PRECISION,
+                    fromScale,
+                    MAX_SHORT_PRECISION,
+                    toScale,
+                    longTenToNth(fromScale - toScale),
+                    longTenToNth(fromScale - toScale) / 2);
+        }
+
+        return rescale(value, fromScale, toScale);
```

Comment:
Could we just delete to `shortToShortCast` in both cases and make `rescaleWithRounding` essentially an alias?

Reply:
Additionally, I have renamed the function `rescaleWithRounding` to `rescaleAndRoundHalfUp` to better reflect the operation of the function.